N1_IntroToMonteCarlo.ipynb

```diff
@@ -0,0 +1,807 @@
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## Programming Exercise 1: Markov Chain\n",
+        "\n",
+        "Below is a partially completed program that simulates the Markov Chain model shown in the figure above.  \n",
+        "\n",
+        "a) Complete the program, and report the average number of steps it takes to change to state $A$ when starting in state $E$.  Include the uncertainty (i.e. [standard error](https://en.wikipedia.org/wiki/Standard_error)) in your reported number.  You should make additional code cells in your notebook to keep track of your data and present your results.\n",
+        "\n",
+        "\n",
+        "\n",
+        "---\n",
+        "Here is the transition probability table:\n",
+        "\n",
+        "State Change | Probability \n",
+        "--- | ---\n",
+        "EE | 0.3\n",
+        "EA | 0.7\n",
+        "AE | 0.4\n",
+        "AA | 0.6"
+      ],
+      "metadata": {
+        "id": "0nlkg2yqJngc"
+      }
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 2,
+      "metadata": {
+        "id": "OZqVToXUx1qG"
+      },
+      "outputs": [],
+      "source": [
+        "# Here are some variables to get you started\n",
+        "\n",
+        "# Decide how many steps to make the simulation run\n",
+        "num_steps = 100\n",
+        "\n",
+        "# Define a starting state: \n",
+        "current_state = 'E' \n",
+        "\n",
+        "# Define an ending state:\n",
+        "end_state = 'A'"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# Decide on a probability distribution function for the simulation \n",
+        "# (uncomment one of the probability lines in the function): \n",
+        "\n",
+        "def get_probability(): \n",
+        "  # This function returns a probability value when called\n",
+        "\n",
+        "  # --> This is a sample from a uniform distribution from numpy\n",
+        "  \n",
+        "  #prob = np.random.uniform(0, 1, 1)\n",
+        "\n",
+        "  # --> This is a sample from a normal distribution from numpy\n",
+        "  # --> It is centered at 0.5 to match the default uniform distribution\n",
+        "  \n",
+        "  #prob = np.random.normal(loc=0.5, size=1)\n",
+        "  \n",
+        "  return prob"
+      ],
+      "metadata": {
+        "id": "O4Wkbgt-dkCC"
+      },
+      "execution_count": 1,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# Now we run the simulation\n",
+        "\n",
+        "for step in range(num_steps):\n",
+        "  p = get_probability()\n",
+        "\n",
+        "  # --- HERE IS THE STATE MACHINE CODE --- #\n",
+        "\n",
+        "  # --> The first logical case is written for you\n",
+        "\n",
+        "  # If the state is currently 'E' \n",
+        "  # and transition probability is less than or equal to 0.3\n",
+        "  if current_state == 'E' and p<=0.3:\n",
```

Comment:
I think these if statements should maybe have a `continue` in them? As they are currently, the state is changed and then the other if statements are processed before the step is increased, so the results may be artificially short. What do you think?

Reply:
Actually, the `continue` command alone wouldn't be enough. You probably need to have one large if-else to check if there is a current state of E or A and then do the rest inside that.

Reply:
Thank you for your comment.  I have updated three of the `if` statements to `elif` to adjust for the "continue" statement you suggested.  Because the final `if` statement occurs at the same level as the state change `if` statements, it covers the case of a current state E or A.